### PR TITLE
docs: remove spinner + reduce load time + add rowDataMapper - PTL-1517

### PR DESCRIPTION
**Pull request by MrBrunoWolff**
This will:
- Add missing `:rowDataMapper` attribute
- Fix `:request` attribute
- Remove loading + reduce delay... working fine with 500ms, no need for extra "noise"/distraction 
- Small documentation tweaks

Do the changes you have made apply to both Current and Previous versions?
No

Have you done a trial build to check all new or changed links?
Yes

Is there anything else you would like us to know?
No

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js
[the remark this answers is not in the document]
Comment (MrBrunoWolff) [reply docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js:1]: it never failed for me.. @matteematt can you re-check this branch? let me know if it fails

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_02_datasources.mdx [hunk 1]
```diff
@@ -208,6 +208,21 @@ Some **resource** attributes are `DATASERVER` or `REQUEST_SERVER` specific, whil
...
 
+</td>
+    </tr>
+    <tr>
+      <td><code>:rowDataMapper</code></td>
+      <td><code>Function</code></td>
+      <td>Function to map the row data before it is sent to the grid. The function should return an array of objects, where each object represents a row in the grid.</td>
+<td>
+
+```html
```
[the remark this answers is not in the document]
Comment (MrBrunoWolff) [reply docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_02_datasources.mdx:209]: see https://github.com/genesiscommunitysuccess/docs/pull/2042#discussion_r1871530142

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_02_datasources.mdx [hunk 2]
```diff
@@ -148,74 +149,109 @@ Some **resource** attributes are `DATASERVER` or `REQUEST_SERVER` specific, whil
...
+<grid-pro-server-side-datasource resource-name="DATASERVER_OR_REQUEST_SERVER_NAME" view-number="2">
+```
+</td>
+    </tr>
+    <tr>
+      <td><code>:rowDataMapper</code></td>
+      <td><code>Function</code></td>
+      <td>Function to map the row data before it is sent to the grid. The function should return an array of objects, where each object represents a row in the grid.</td>
+<td>
+```html
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_02_datasources.mdx:209]: You can add that to the properties table you created too

**Review by matteematt — COMMENTED** [review]
Looks great after the new changes, just left one comment. Also I tested and 500 seems ok now. Happy to merge that, and can adjust later if feedback recommends

**Comment by MrBrunoWolff** [comment]
@matteematt should be ok now, had a conflict with a commit from `preprod`

**Comment by jay-taylerson** [comment]
@MrBrunoWolff @matteematt please let us know if this is approval/merge worthy

**Review by matteematt — APPROVED** [review]

**Comment by jay-taylerson** [comment]
Looks OK to me. The one thing which might be useful, and keen for your thoughts on... but in server for placeholders we generally use a form of brackets to denote.

So instead of `resource-name="DATASERVER_OR_REQUEST_SERVER_NAME"` it would be `resource-name="<DATASERVER_OR_REQUEST_SERVER_NAME>"`

I realise <> doesn't work well in client docs, should we use [] ?

`resource-name="[DATASERVER_OR_REQUEST_SERVER_NAME]"`

@matteematt @MrBrunoWolff

**Comment by jay-taylerson** [comment]
Maybe not, happy to be pushed back on this FYI.

**Comment by MrBrunoWolff** [comment]
that sounds interesting and something that got me thinking yesterday, "wildcard scenarios" 

I'd leave that for a separate PR, that would scan all the pages in one go, but definitely a "next thing"

**Comment by matteematt** [comment]
> Looks OK to me. The one thing which might be useful, and keen for your thoughts on... but in server for placeholders we generally use a form of brackets to denote.
> 
> So instead of `resource-name="DATASERVER_OR_REQUEST_SERVER_NAME"` it would be `resource-name="<DATASERVER_OR_REQUEST_SERVER_NAME>"`
> 
> I realise <> doesn't work well in client docs, should we use [] ?
> 
> `resource-name="[DATASERVER_OR_REQUEST_SERVER_NAME]"`
> 
> @matteematt @MrBrunoWolff

Interesting suggstion. I think we've taken a bit of a different approach in our docs. Rather than showing a placeholder value we've tended to show concrete examples. I quite like this as I think it aids understanding for people who are new to the concept. Perhaps there are times where we are being generic though, in that case I think we should add some symbols as suggested to make it clear they're placeholders and you can't copy and paste exactly.

**Pull request merged**